src/Libraries/Microsoft.Extensions.AI.Abstractions/Utilities/AIJsonUtilities.Defaults.cs

```diff
@@ -83,4 +86,60 @@ private static JsonSerializerOptions CreateDefaultOptions()
     [JsonSerializable(typeof(Embedding<double>))]
     [JsonSerializable(typeof(AIContent))]
     private sealed partial class JsonContext : JsonSerializerContext;
+
+    private sealed class JsonStringBooleanConverter : JsonConverter<bool>
+    {
+        public override bool Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
+        {
+            bool? boolResult = null;
+
+            if (reader.TokenType == JsonTokenType.String)
+            {
+#if NET
+                static int GetValueLength(ref Utf8JsonReader reader)
+                    => reader.HasValueSequence
+                        ? checked((int)reader.ValueSequence.Length)
+                        : reader.ValueSpan.Length;
+
+                const int StackallocByteThreshold = 256;
+                const int StackallocCharThreshold = StackallocByteThreshold / 2;
+
+                int bufferLength = GetValueLength(ref reader);
+                char[]? rentedBuffer = null;
```

Comment:
Valid boolean values are at most 5 characters long. You can avoid pooling buffers if you just check the length of the input before stack allocating a buffer of length 5.

Reply:
I don't know if we care, but if there was whitespace padding around the true/false, it could be longer than 5 chars. bool.Parse trims and thus will happily successfully parse e.g. "                    true                           ".